docs/book.md

```diff
@@ -0,0 +1,299 @@
+
+Once these bespoke HTTP client configurations are built, both they and their dependencies are now added to the grab bag of dependency versions that must be maintained through the life of any given piece of infrastructure. This presents hidden barriers for upgrading all dependencies, as the possibility of transitive dependency conflicts increase as dependency trees become deeper.
+
+Installation
+============
+
+Guardrail is available as a modular core, with both [sbt]() and [Maven]() integration. The core can also be run as a stand-alone [CLI]() application, with full support for all features.
+
+Sample API specification
+========================
+
+The following is a complete, annotated OpenAPI specification file:
+
+    swagger: "2.0"                              # Which version of the OpenAPI/Swagger specification we are following
+    info:                                       # Primarily for consumption by documentation generation tools
+      title: My Service
+      version: 0.1.0
+    host: localhost:1234                        # Default host (and optional port) to connect to for generated clients
+    schemes:
+      - http
+    paths:                                      # All HTTP paths are direct children of the `paths` field
+
+      /user/{id}:                               # Paths can have variable patterns in paths
+
+        get:                                    # HTTP method
+
+          operationId: getUser                  # Friendly name, ends up as the function name (in clients and servers)
+
+          x-scala-package: users                # Relative package for this client to live in. For convenience, the
+                                                # last package parameter is turned into a class name for clients and
+                                                # servers. In this case, `UsersClient`.
+
+          parameters:                           # All parameters (including path parameters) are listed here.
+
+          - name: id                            # The field name (case matters!), used to both identify the correct
+                                                # field to match, as well as generate a best-guess idiomatic Scala
+                                                # parameter name.
+
+            in: path                            # Where to look for the parameter
+
+            required: true                      # Required fields cannot be missing. `required: false` fields are
+                                                # represented as `Option[T]`
+
+            type: string                        # One of the primitive types supported in the OpenAPI specification.
+                                                # FIXME: Include a link to the specification for types and formats
+
+            x-scala-type: CustomString          # Escape hatch to explicitly introduce a custom type. This is an
```

Comment:
Would this comment be better as just a link to a section of the Guardrail book that describes these more advanced features?

Reply:
Ah, yes, that would be helpful.